### ISSUE TITLE
building-telemtry exercise instructions are vague

### ISSUE THREAD

**Issue by gmf002** [exercism/csharp#1832]
In step 2, the instructions say that the GetTelemetry method should return false the provided serial number is less than the previous serial number.

In step 3, the instructions say that if GetTelemetry returns false to return "no data".  However, the unit test expects that "no data" should be returned if the car has not been driven.  Based on the instruction in step 2, GetTelemetry returns true in this case.

Either step 2 should be updated to say reflect less than or equal to the previous serial number, or step 3 should be updated to reflect that "no data" should also be returned if car has not been driven.

Based on the units tests, it appears that step 3 should be where the clarification goes.

**Comment by ErikSchierboom** [comment]
Looking at the exemplar solution, it has the following code:

```csharp
if (serialNum > latestSerialNum)
{
    latestSerialNum = serialNum;
    batteryPercentage = this.batteryPercentage;
    distanceDrivenInMeters = this.distanceDrivenInMeters;
    return true;
}
else
{
    serialNum = latestSerialNum;
    batteryPercentage = -1;
    distanceDrivenInMeters = -1;
    return false;
}
```

So I think the description of step 2 needs to be updated.

**Comment by gmf002** [comment]
@ErikSchierboom  I hadn't uploaded my solution yet when i wrote the issue, so seeing that now makes more sense.   So, the instructions for step 2 should be updated, and a new unit test for that condition probably should be added.

**Comment by gmf002** [comment]
Actually, looking at what I needed to do to get my code to pass the tests, it may still be step 3 (based purely on the unit tests).

My step two method has

```csharp
        if (serialNum < latestSerialNum)
        {
            serialNum = latestSerialNum;
            batteryPercentage = -1;
            distanceDrivenInMeters = -1;
            return false;
        }

        latestSerialNum = serialNum;
        batteryPercentage = this.batteryPercentage;
        distanceDrivenInMeters = this.distanceDrivenInMeters;
        return true;
```

which is equivalent to the exemplar solution.

What I needed to do to get the step 3 method to pass the tests was

```csharp
        var dataAvailable =
            car.GetTelemetryData(ref serialNum, out var batteryPercentage, out var distanceDrivenInMeters);

        return dataAvailable == false || batteryPercentage == 100
            ? "no data"
            : $"usage-per-meter={(100 - batteryPercentage) / distanceDrivenInMeters}";
```

So, either the instructions or test for step 3 are missing requirements as it makes no mention of the need for the car to be started, or the unit test is wrong.

You could instead require step 2 to return false if the battery was at 100%.  That would make some sense because if the car is not started, it can't very well send telemetry data.

**Comment by ErikSchierboom** [comment]
> You could instead require step 2 to return false if the battery was at 100%. That would make some sense because if the car is not started, it can't very well send telemetry data.

I think we _have_ to add this clause, as other the `(100 - batteryPercentage) / distanceDrivenInMeters` would result in a divide by zero exception. The exemplar solution uses `distanceDrivenInMeters == 0` (instead of `batteryPercentage == 100`), which intuitively makes slightly more sense to me.

**Comment by gmf002** [comment]
Talking through this with you, I agree distance driven makes more sense.  So, the instructions for step 3 should be updated to state something like:

... If GetTelemetryData() returns false then this routine should return "no data", _or the car has not been driven_. ...

Alternatively, perhaps "_or battery usage per meter is 0_".  The first one makes more sense to me based on the implementation from the exemplar solution.

I can submit a pull request if that is agreeable.

**Comment by ErikSchierboom** [comment]
Yep that works for me. I'd probably rephrase to:

If GetTelemetryData() returns false or the car has not been driven, then this routine should return "no data", . ...

**Comment by gmf002** [comment]
I submitted a pull request #1835 with your revised phrasing.

**Comment by ErikSchierboom** [comment]
Closed by #1835
